zeroeq/http/server.cpp

```diff
@@ -68,58 +98,71 @@ class Server::Impl : public detail::Sender
         initURI();
     }
 
-    bool subscribe( servus::Serializable& serializable )
+    bool handlePUT( servus::Serializable& serializable )
     {
         const auto func = [&serializable]( const std::string& json )
             { return serializable.fromJSON( json ); };
-        return subscribe( serializable.getTypeName(), func );
+        return handlePUT( serializable.getTypeName(), serializable.getSchema(),
+                          func );
     }
 
-    bool subscribe( std::string event, const PUTPayloadFunc& func )
+    bool handlePUT( std::string event, const std::string& schema,
+                    const PUTPayloadFunc& func )
     {
-        event = _toLower( event );
-        if( _subscriptions.count( event ) != 0 )
+        event = _toLower( _replaceAll( event, "/", "::" ));
+        if( _put.count( event ) != 0 )
             return false;
 
-        _subscriptions[ event ] = func;
+        _put[ event ] = func;
+        if( !schema.empty( ))
+            _schemas[ event ] = schema;
```

Comment:
At least you should do a string comparison then. What does it mean to register GET and PUT with different schemas otherwise?